### PR TITLE
Run pre-commit on all files periodically

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

In light of all the changes originating from running more complex pre-commit checks (`isort` in particular), I think it would be useful to occasionally run the `pre-commit` checks on the entire codebase to ensure that we remain consistent. This is because the pre-commit.ci bot only runs on the files changed by a PR not the entire codebase.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by nstarman** [comment]
I think this a good idea. My only Q is whether this should be run weekly or monthly, not daily. Bikeshedding, but we will eventually stabilize the pre-commit and it runs on every PR, so  it won't make changes to the rest of the code very often.

**Comment by pllim** [comment]
We already have a workflow for weekly, but not monthly.

**Comment by WilliamJamieson** [comment]
Swapped it to weekly, since that one already exists.

**Comment by pllim** [comment]
Syntax error? `ERROR: InvocationError for command /home/runner/work/astropy/astropy/.tox/codestyle/bin/pre-commit run --all-files -v (exited with code 1)`

**Comment by WilliamJamieson** [comment]
> Syntax error? `ERROR: InvocationError for command /home/runner/work/astropy/astropy/.tox/codestyle/bin/pre-commit run --all-files -v (exited with code 1)`

CI it just failed:
```
isort (python)...........................................................Failed
- hook id: isort
- duration: 2.4s
- files were modified by this hook

Fixing /home/runner/work/astropy/astropy/astropy/io/fits/setup_package.py
Skipped 287 files
```

**Comment by pllim** [comment]
Hmm maybe rebase? I just merged some isort PRs.

**Comment by WilliamJamieson** [comment]
> Hmm maybe rebase? I just merged some isort PRs.

This is a legitimate error, in fact its exactly the sort of thing adding this testing was supposed to catch.

@nstarman should we fix the error here or in a new PR.

**Comment by WilliamJamieson** [comment]
We see the error here on a merge to main too: https://results.pre-commit.ci/run/github/2081289/1663679465.PZ8WkR2-SeSee790SwLM1Q

**Comment by pllim** [comment]
Might as well fix it here if it is not too much trouble?

**Comment by pllim** [comment]
Or disable isort for `io.fits`.

**Comment by WilliamJamieson** [comment]
> Might as well fix it here if it is not too much trouble?

Done

**Comment by pllim** [comment]
I am going to cancel the irrelevant CI. Only that one job you added matters.

**Comment by WilliamJamieson** [comment]
Per this comment, https://github.com/astropy/astropy/pull/13695#issuecomment-1252380978. This CI job may not even be needed.

**Comment by pllim** [comment]
Hmm. Well, if not needed, then feel free to close without merge.

The log for this PR shows the changelog stuff is skipped, is that intentional? https://github.com/astropy/astropy/actions/runs/3090630025/jobs/4999680067

**Comment by WilliamJamieson** [comment]
I've never seen those pre-commit checks actually run. For example, #12579 has a changelog fragment, but those checks get skipped: https://results.pre-commit.ci/run/github/2081289/1663617810.Xj7vLw7KRRqy15k20JwjdQ.

**Comment by pllim** [comment]
@saimn or @Cadair , do you what those change log precommit checks are about?

**Comment by saimn** [comment]
it checks that only allowed categories are used and `other` is not used in sub-packages, not sure why it doesn't run thoug.

**Comment by saimn** [comment]
actually the skip is normal, it's because it's not running a real command, just using the file detection/exclusion. If a file doesn't match the exclude patterns then it will fail.

File: astropy/io/fits/setup_package.py [hunk 1]
```diff
@@ -6,9 +6,10 @@
 from glob import glob
 
 import numpy
-from extension_helpers import get_compiler, pkg_config
 from setuptools import Extension
 
+from extension_helpers import get_compiler, pkg_config
```
[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/io/fits/setup_package.py:11]: Classic

**Comment by nstarman** [comment]
> > Hmm maybe rebase? I just merged some isort PRs.
> 
> This is a legitimate error, in fact its exactly the sort of thing adding this testing was supposed to catch.
> 
> @nstarman should we fix the error here or in a new PR?

I'd say fix it here! As you say this exactly the sort of thing adding this testing was supposed to catch.

**Review by nstarman — APPROVED** [review]
Thanks @WilliamJamieson. Self-evidently a useful addition.

**Comment by eerovaher** [comment]
There are quite a few tests in `pre-commit` that do not have to be ran periodically. For example, if none of the pull requests introduce trailing whitespaces then checking the whole codebase periodically will not reveal any new instances. Would it be worth it to come up with a smaller list of tasks for the periodic `pre-commit` check?

**Comment by pllim** [comment]
So, the fits fix is already in via https://github.com/astropy/astropy/pull/13723 . Should we close this without merge? The conversation is confusing me.

**Comment by WilliamJamieson** [comment]
I think we should close for now, maybe we should try to commit a pre-commit change that will effect files not changed in the PR and see if that triggers a failure of the pre-commit.ci?

**Pull request closed without merging**